crates/solvers/src/infra/observe/mod.rs

```diff
@@ -0,0 +1,34 @@
+use {
+    crate::domain::{auction, solution},
+    chrono::Utc,
+};
+
+pub mod metrics;
+
+pub fn solve(auction: &auction::Auction) {
```

Comment:
The differentiation between the `observe` module and `metrics` does not seem worth it to me. I think we only have an `observe` module in the `driver` code because that was supposed to handle logging and metrics but that turned out to be not great.
I think we could simply expose an `infra::metrics` module and make these functions members of `Metrics`.